core/error.rs

```diff
@@ -236,25 +236,82 @@ impl JsError {
         frames_v8.and_then(|a| a.try_into().ok());
 
       // Convert them into Vec<JsStackFrame>
-      let frames: Vec<JsStackFrame> = match frames_v8 {
+      let mut frames: Vec<JsStackFrame> = match frames_v8 {
         Some(frames_v8) => serde_v8::from_v8(scope, frames_v8.into()).unwrap(),
         None => vec![],
       };
+
+      let state_rc = JsRuntime::state(scope);
+      let state = state_rc.borrow();
+
+      // When the stack frame array is empty, but the source location given by
+      // (script_resource_name, line_number, start_column + 1) exists, this is
+      // likely a syntax error. For the sake of formatting we treat it like it
+      // was given as a single stack frame.
+      if frames.is_empty() {
+        let script_resource_name = msg
+          .get_script_resource_name(scope)
+          .and_then(|v| v8::Local::<v8::String>::try_from(v).ok())
+          .map(|v| v.to_rust_string_lossy(scope));
+        let line_number: Option<i64> =
+          msg.get_line_number(scope).and_then(|v| v.try_into().ok());
+        let column_number: Option<i64> = msg.get_start_column().try_into().ok();
+        if let (Some(f), Some(l), Some(c)) =
+          (script_resource_name, line_number, column_number)
+        {
+          // V8's column numbers are 0-based, we want 1-based.
+          let c = c + 1;
+          if let Some(source_map_getter) = &state.source_map_getter {
+            let (f, l, c, _) = apply_source_map(
+              f,
+              l,
+              c,
+              &mut HashMap::new(),
+              source_map_getter.as_ref(),
+            );
+            frames =
+              vec![JsStackFrame::from_location(Some(f), Some(l), Some(c))];
+          } else {
+            frames =
+              vec![JsStackFrame::from_location(Some(f), Some(l), Some(c))];
+          }
+        }
+      }
+
+      let mut source_line = None;
+      let mut source_line_frame_index = None;
+      if let Some(source_map_getter) = &state.source_map_getter {
+        for (i, frame) in frames.iter().enumerate() {
+          if let (Some(file_name), Some(line_number)) =
+            (&frame.file_name, frame.line_number)
+          {
+            if !file_name.starts_with("deno:") {
```

Comment:
I think we should also check for file names starting with `[deno:` (the ones create by `located_script_name!` macro)